### DISCUSSION TITLE
performance degradation in keycloak 18/19/20

### DISCUSSION THREAD

**Discussion by kkcmadhu** [keycloak/keycloak#17020] [Q&A]
I see study performance degradation from keycloak 18 onwards with token/ roles/ search and other endpoints.

With keycloak 17 on wildfly the performance seems to be better..
Last time i attempted to upgrade to keycloak 18, it was rejected by one perf test team stating there is quite some degradation in performance. 

I see similar issues now in keycloak 20 too ( though this seems to be better than keycloak 18)..

Here is a snapshot of my perf test results using jmeter..
**Keycloak 20**

![kc_20](https://user-images.githubusercontent.com/997582/218371769-a0828995-7119-45c0-b0aa-c00731fa8e3d.png)

**Keycloak 17**
![kc_17](https://user-images.githubusercontent.com/997582/218371723-793f1aca-d7cb-447b-bccf-5f5be387b588.png)

Summary

<html>
<body>
<span><span class="ui-provider bus but c d e f g h i j k l m n o p q r s t buu buv w x y z ab ac ae af ag ah ai aj ak" dir="ltr"><span><span style="font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, system-ui, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;, &quot;Segoe UI Web&quot;, sans-serif; font-size: 14px; white-space: pre-wrap;">
<span><p>&nbsp;</p>

  | KC_17 3hrs  (ms) | KC_20 3hrs (ms) |  
-- | -- | -- | --
GET_USER | 385 | 585 | -200
GET_USER_ROLES | 390 | 684 | -294
LOGIN(OpenID) | 917 | 1076 | -159
LOGOUT(OpenID) | 299 | 319 | -20
SEARCH USER | 931 | 1097 | -166

<p>&nbsp;</p></span>
</span>
</span></span></span>
</body>
</html>
I suspect the login call has underwent some kind ofdegradation.. not sure though.. any one else observed similar issues in keycloak 18/19/20 on quarkus?

It is worth while noting that i saw similar degradation between keycloak 13 and keycloak 15 too :(

**Comment by codespearhead** [comment]
Same issue as [1]. Could you please [close this discussion](https://docs.github.com/en/discussions/managing-discussions-for-your-community/managing-discussions#closing-a-discussion)?

[1] https://github.com/keycloak/keycloak/discussions/29126
